src/core/thread.d

```diff
@@ -2239,8 +2244,14 @@ version( Windows )
  */
 extern (C) void thread_detachThis() nothrow @nogc
 {
+    Thread.slock.lock_nothrow();
+    scope(exit) Thread.slock.unlock_nothrow();
```

Comment:
Ideally, these changes should be in the separate commit.

Reply:
Agreed but I could not pass my test program without fixing these as well. Perhaps I should open two bugs and submit fixes for two.

Reply:
Why not just add extra commit in this PR? Make the first commit to add locks, and the second commit (which includes `Fixes issue ...` entry) with `t.setThis(null)` and doc change?